### PR TITLE
Handle CORS

### PR DESCRIPTION
**Pull request by andela-ssunday**

### CODE REVIEW

File: kit/service/transport.go [hunk 1]
```diff
@@ -98,6 +99,18 @@ func NewRouter(
 		w.Write([]byte("ok"))
 	})
 
+	//handle CORS
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```
Comment (vood) [inline kit/service/transport.go]: Please remove this. We don't need it anymore.

File: kit/service/transport.go [hunk 2]
```diff
@@ -131,7 +132,7 @@ func (r *Router) MountRoute(route Route) *Router {
 		_, handler = newrelic.WrapHandle(r.newrelicApp, route.GetPath(), handler)
 	}
 
-	r.prefixMux.Methods(route.GetMethod()).Path(route.GetPath()).Handler(handler)
+	r.prefixMux.Methods(route.GetMethod()).Path(route.GetPath()).Handler(handlers.CORS()(handler))
```
Comment (vood) [inline kit/service/transport.go]: By default this lib does not allow PUT requests:
```
defaultCorsMethods = []string{"GET", "HEAD", "POST"}
defaultCorsHeaders = []string{"Accept", "Accept-Language", "Content-Language", "Origin"}
```

We also need to allow our custom headers: 'X-Connect-Client-Type' and 'X-Connect-Client-Version'

Comment (ibrt) [inline kit/service/transport.go]: Can we make this an option on the Route object that's off by default?

File: kit/service/transport.go [hunk 3]
```diff
@@ -131,7 +132,10 @@ func (r *Router) MountRoute(route Route) *Router {
 		_, handler = newrelic.WrapHandle(r.newrelicApp, route.GetPath(), handler)
 	}
 
-	r.prefixMux.Methods(route.GetMethod()).Path(route.GetPath()).Handler(handler)
+	headers := handlers.AllowedHeaders([]string{"X-Connect-Client-Type", "X-Connect-Client-Version", "Origin", "Content-Type"})
```
Comment (vood) [inline kit/service/transport.go]: Are you removing "Accept", "Accept-Language", "Content-Language" on purpose?

Comment (andela-ssunday) [inline kit/service/transport.go]: No its not removed. The AllowedHeaders method performs an append https://github.com/gorilla/handlers/blob/master/cors.go#L168

**Pull request merged**